### PR TITLE
Geo field index

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ qdrant-client = "1.4.0"
 rand = "0.8.5"
 tokio = { version = "1.29.1", features = ["full"] }
 uuid = { version = "1.4.0", features = ["serde", "v4"] }
-
+serde_json = "~1.0"
```
Comment (timvisee) [inline Cargo.toml:19]: Please sort the list of dependencies alphabetically :smile:

File: src/common.rs [hunk 2]
```diff
@@ -2,31 +2,57 @@ use core::option::Option;
...
     if float_payloads {
         payload.insert(FLOAT_PAYLOAD_KEY, rand::thread_rng().gen_range(-1.0..1.0));
     }
+    if geo_payloads {
+        payload.insert(
+            GEO_PAYLOAD_KEY,
+            json!(
+            {
+                "lon": BERLIN.lon + rand::thread_rng().gen_range(-GEO_RADIUS..GEO_RADIUS),
+                "lat": BERLIN.lat + rand::thread_rng().gen_range(-GEO_RADIUS..GEO_RADIUS),
```
Comment (timvisee) [inline src/common.rs:44]:
```suggestion
                "lon": BERLIN.lon + rand::thread_rng().gen_range(-GEO_RADIUS..=GEO_RADIUS),
                "lat": BERLIN.lat + rand::thread_rng().gen_range(-GEO_RADIUS..=GEO_RADIUS),
```

File: src/common.rs [hunk 3]
```diff
@@ -68,6 +94,23 @@ pub fn random_filter(keywords: Option<usize>, float_payloads: bool) -> Option<Fi
...
+        filter.must.push(
+            FieldCondition {
+                key: GEO_PAYLOAD_KEY.to_string(),
+                r#match: None,
+                range: None,
+                geo_bounding_box: None,
+                geo_radius: Some(GeoRadius {
+                    center: Some(BERLIN),
+                    radius: GEO_RADIUS as f32,
+                }),
```
Comment (timvisee) [inline src/common.rs:108]: Should we randomize the center and range here? With this configurations, all points are always included.

**Review by timvisee — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (IvanPleshkov) [reply Cargo.toml]: done

File: src/common.rs (hunk 3 above)
Comment (timvisee) [reply src/common.rs:108]: > Not all points, square is larger

Good point. Thanks for clarifying.

**Review by timvisee — APPROVED** [review]

**Pull request closed without merging**